### PR TITLE
base file for unit-testes for unary operators

### PR DESCRIPTION
**Pull request by bszmelcz**
Add file for unit tests for all unary operators

### CODE REVIEW

File: ngraph/test/type_prop/unary_base.cpp
Comment (jdanieck) [inline ngraph/test/type_prop/unary_base.cpp:156]: You can simplify test instantiation with use of testing::Combine.

_const std::vector<InputType> precisions = { InputType::f32, InputType::f16, InputType::i32, InputType::i16 };
const std::vector<Shape> shapes = { Shape{2, 2}, Shape{100, 2, 50}, Shape{5, 243, 51}};
INSTANTIATE_TEST_CASE_P(type_prop,
                        BasicParamInference,
                        testing::Combine(testing::ValuesIn(precisions), testing::ValuesIn(shapes)),
                        BasicParamInference::getTestCaseName);_

File: ngraph/test/type_prop/unary_base.cpp
Comment (jdanieck) [inline ngraph/test/type_prop/unary_base.cpp:144]: Did you try to parametrize further your tests with operation type (e.g. op::Acos) ? There is [Typed Tests](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#typed-tests) feature available, but I didn't try it myself yet.

Comment (bszmelcz) [reply ngraph/test/type_prop/unary_base.cpp:144]: Did not try, but if is preferred I can check that

File: ngraph/test/type_prop/unary_base.cpp
Comment (bszmelcz) [reply ngraph/test/type_prop/unary_base.cpp:156]: Great, thanks! I just copied it from the example.

File: ngraph/test/type_prop/unary_base.cpp
Comment (pelszkow) [reply ngraph/test/type_prop/unary_base.cpp:47]: Some hint: https://godbolt.org/z/sazv66

**Review by jdanieck — COMMENTED** [review]
@bszmelcz I don't know if it is final version or not, but I'm finishing my last day before vacations so I'll leave a comment. 

First of all please read and follow [OpenVINO PR review good practices] (https://github.com/openvinotoolkit/openvino/wiki/Contribute#pr-review-good-practices) - it will allow us to make PR process more productive. 

Second thing is that I see you used Typed Tests, thats nice, but I hoped we can combine Typed Tests with Parameterized tests (what you had before) Please try to produce "Typed Parameterized" version. 

If it is not possible then I like better 'Typed' version because it gives us opportunity to easily add other unary operators. So if we stick to this one then try to separate those block scope fragments into separate test cases.

_TYPED_TEST_P(UnaryOperator, incompatible_input_type_Shape1)
{
        const auto param = std::make_shared<op::Parameter>(element::boolean, Shape{100, 2, 50});
        ASSERT_THROW(std::make_shared<TypeParam>(param), ngraph::NodeValidationFailure);
}

TYPED_TEST_P(UnaryOperator, incompatible_input_type_Shape2)
    {
        const auto param = std::make_shared<op::Parameter>(element::boolean, Shape{40, 17, 50});
        ASSERT_THROW(std::make_shared<TypeParam>(param), ngraph::NodeValidationFailure);
    }
}_

File: ngraph/test/type_prop/unary_ops.cpp
Comment (ilyachur) [inline ngraph/test/type_prop/unary_ops.cpp:40]: Looks like we grouped several test cases in one bug test case. Can we split them and use separate test cases?

File: ngraph/test/type_prop/unary_ops.cpp
Comment (ilyachur) [inline ngraph/test/type_prop/unary_ops.cpp:70]: Can we add tests for partial dynamic shapes? I mean something like that `{1, dynamic(), 3}`

File: ngraph/test/type_prop/unary_ops.cpp
Comment (bszmelcz) [reply ngraph/test/type_prop/unary_ops.cpp:40]: sure

**Comment by bszmelcz** [comment]
> @bszmelcz I don't know if it is final version or not, but I'm finishing my last day before vacations so I'll leave a comment.
> 
> First of all please read and follow [OpenVINO PR review good practices](https://github.com/openvinotoolkit/openvino/wiki/Contribute#pr-review-good-practices) - it will allow us to make PR process more productive.
> 
> Second thing is that I see you used Typed Tests, thats nice, but I hoped we can combine Typed Tests with Parameterized tests (what you had before) Please try to produce "Typed Parameterized" version.
> 
> If it is not possible then I like better 'Typed' version because it gives us opportunity to easily add other unary operators. So if we stick to this one then try to separate those block scope fragments into separate test cases.
> 
> _TYPED_TEST_P(UnaryOperator, incompatible_input_type_Shape1)
> {
> const auto param = std::make_sharedop::Parameter(element::boolean, Shape{100, 2, 50});
> ASSERT_THROW(std::make_shared(param), ngraph::NodeValidationFailure);
> }
> 
> TYPED_TEST_P(UnaryOperator, incompatible_input_type_Shape2)
> {
> const auto param = std::make_sharedop::Parameter(element::boolean, Shape{40, 17, 50});
> ASSERT_THROW(std::make_shared(param), ngraph::NodeValidationFailure);
> }
> }_

From what I have read, combinig those tests is not supported at the moment and it is not planned to be. So as u said i will stick to typed implementation and seperate block fragments.

**Review by lazarevevgeny — APPROVED** [review]
LGTM. But please, get approval from @jdanieck before merge

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
Please rebase your PR to the latest master

**Review by jdanieck — APPROVED** [review]

**Pull request merged**
